chain/default_syncer.go

```diff
@@ -33,6 +37,25 @@ var (
 
 var logSyncer = logging.Logger("chain.syncer")
 
+// SyncMode is the
+type SyncMode int
+
+const (
+	// Syncing indicates that the node was started recently and the chain is still
+	// significantly behind the current consensus head.
+	//
+	// See the spec for more detail:
+	// https://github.com/filecoin-project/specs/blob/master/sync.md#syncing-mode
+	Syncing SyncMode = iota
+	// CaughtUp indicates that the node has caught up with consensus head and as a
+	// result can restrict which new blocks are accepted to mitigate consensus
+	// attacks.
+	//
+	// See the spec for more detail:
+	// https://github.com/filecoin-project/specs/blob/master/sync.md#caught-up-mode
+	CaughtUp
```

Comment:
When do we enter this mode? I only see logic for `Syncing` mode. Is this to be done in a follow on or has it not been addressed because this is on ice?

Reply:
Never, except tests, but that needn't stop this code from landing. However, it cannot close the underlying issue #2782